ultrasonic/src/main/kotlin/org/moire/ultrasonic/fragment/SelectAlbumFragment.kt

```diff
@@ -0,0 +1,1180 @@
+import android.widget.AdapterView.OnItemClickListener
+import android.widget.AdapterView.OnItemLongClickListener
+import android.widget.ImageView
+import android.widget.ListView
+import android.widget.TextView
+import androidx.fragment.app.Fragment
+import androidx.navigation.Navigation
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout.OnRefreshListener
+import java.security.SecureRandom
+import java.util.Collections
+import java.util.LinkedList
+import java.util.Random
+import org.koin.java.KoinJavaComponent
+import org.moire.ultrasonic.R
+import org.moire.ultrasonic.data.ActiveServerProvider.Companion.isOffline
+import org.moire.ultrasonic.domain.MusicDirectory
+import org.moire.ultrasonic.fragment.FragmentTitle.Companion.getTitle
+import org.moire.ultrasonic.fragment.FragmentTitle.Companion.setTitle
+import org.moire.ultrasonic.service.MediaPlayerController
+import org.moire.ultrasonic.service.MusicService
+import org.moire.ultrasonic.service.MusicServiceFactory.getMusicService
+import org.moire.ultrasonic.subsonic.DownloadHandler
+import org.moire.ultrasonic.subsonic.ImageLoaderProvider
+import org.moire.ultrasonic.subsonic.NetworkAndStorageChecker
+import org.moire.ultrasonic.subsonic.ShareHandler
+import org.moire.ultrasonic.subsonic.VideoPlayer
+import org.moire.ultrasonic.util.AlbumHeader
+import org.moire.ultrasonic.util.CancellationToken
+import org.moire.ultrasonic.util.Constants
+import org.moire.ultrasonic.util.EntryByDiscAndTrackComparator
+import org.moire.ultrasonic.util.FragmentBackgroundTask
+import org.moire.ultrasonic.util.Util
+import org.moire.ultrasonic.view.AlbumView
+import org.moire.ultrasonic.view.EntryAdapter
+import org.moire.ultrasonic.view.SongView
+import timber.log.Timber
+
+/**
+ * Displays a group of playable media from the library, which can be an Album, a Playlist, etc.
+ */
+class SelectAlbumFragment : Fragment() {
+
+    private var refreshAlbumListView: SwipeRefreshLayout? = null
+    private var albumListView: ListView? = null
+    private var header: View? = null
+    private var albumButtons: View? = null
+    private var emptyView: View? = null
+    private var selectButton: ImageView? = null
+    private var playNowButton: ImageView? = null
+    private var playNextButton: ImageView? = null
+    private var playLastButton: ImageView? = null
+    private var pinButton: ImageView? = null
+    private var unpinButton: ImageView? = null
+    private var downloadButton: ImageView? = null
+    private var deleteButton: ImageView? = null
+    private var moreButton: ImageView? = null
+    private var playAllButtonVisible = false
+    private var shareButtonVisible = false
+    private var playAllButton: MenuItem? = null
+    private var shareButton: MenuItem? = null
+    private var showHeader = true
+    private val random: Random = SecureRandom()
+
+    private val mediaPlayerControllerLazy = KoinJavaComponent.inject(
```

Comment:
Please make sure I did this correctly. I tried to turn the others into `inject()` but it didn't work for Java reasons I don't understand.

Reply:
Yep you did everything fine! You can change the others too, just remove the `.value` part, e.g.:
```
    private val videoPlayer: VideoPlayer by inject()
    ...
    videoPlayer.playVideo(entry)

```
This is necessary because the DI works differently in java and in kotlin, in java it uses Lazy Initialization (with the Lazy<T> class), which receives its value when the `.value` is first accessed. In kotlin, the injected objects don't need to be Lazy<T>, it uses some kotlin magic instead in `by inject()` to return a type T directly...

Reply:
Worked great! I'm not sure what went wrong the last time I tried, probably a simple typo. Anyway, it works now